axelor-account/src/main/java/com/axelor/apps/account/service/batch/BatchAutoMoveLettering.java

```diff
@@ -0,0 +1,482 @@
+          companyPartnerDebitMoveLineList, companyPartnerCreditMoveLineList, reconcileMethodSelect);
+    }
+  }
+
+  protected void reconcileWithMethod(
+      List<MoveLine> debitMoveLines, List<MoveLine> creditMoveLines, int reconcileMethodSelect) {
+    if (CollectionUtils.isNotEmpty(debitMoveLines) && CollectionUtils.isNotEmpty(creditMoveLines)) {
+
+      BigDecimal debitTotalRemaining =
+          debitMoveLines.stream()
+              .map(MoveLine::getAmountRemaining)
+              .reduce(BigDecimal::add)
+              .orElse(BigDecimal.ZERO);
+      BigDecimal creditTotalRemaining =
+          creditMoveLines.stream()
+              .map(MoveLine::getAmountRemaining)
+              .reduce(BigDecimal::add)
+              .orElse(BigDecimal.ZERO);
+
+      Map<MoveLine, BigDecimal> debitRemaining = new HashMap<>();
+      for (MoveLine debitMoveLine : debitMoveLines) {
+        debitRemaining.put(debitMoveLine, debitMoveLine.getAmountRemaining());
+      }
+      for (MoveLine creditMoveLine : creditMoveLines) {
+        BigDecimal creditRemaining = creditMoveLine.getAmountRemaining();
+        for (MoveLine debitMoveLine : debitMoveLines) {
+          BigDecimal debit = debitMoveLine.getDebit();
+          BigDecimal credit = creditMoveLine.getCredit();
+          BigDecimal nextCreditRemaining = creditRemaining.subtract(debit);
+          BigDecimal nextDebitRemaining = debitRemaining.get(debitMoveLine).subtract(credit);
+          if (nextCreditRemaining.signum() < 0 || nextDebitRemaining.signum() < 0) {
+            continue;
+          }
+
+          debitMoveLine = moveLineRepository.find(debitMoveLine.getId());
+          creditMoveLine = moveLineRepository.find(creditMoveLine.getId());
+
+          boolean reconcileByAmount =
+              reconcileMethodSelect
+                      == AccountingBatchRepository.AUTO_MOVE_LETTERING_RECONCILE_BY_AMOUNT
+                  && debit.compareTo(credit) == 0;
+          boolean reconcileByOrigin =
+              reconcileMethodSelect
+                      == AccountingBatchRepository.AUTO_MOVE_LETTERING_RECONCILE_BY_ORIGIN
+                  && debitMoveLine.getOrigin() != null
+                  && creditMoveLine.getOrigin() != null
+                  && debitMoveLine.getOrigin().equals(creditMoveLine.getOrigin())
+                  && (accountingBatch.getIsPartialReconcile() || debit.compareTo(credit) == 0);
+          boolean reconcileByBalancedMove =
+              reconcileMethodSelect
+                      == AccountingBatchRepository.AUTO_MOVE_LETTERING_RECONCILE_BY_BALANCED_MOVE
+                  && willMakeBalancedMove(creditMoveLine)
+                  && willMakeBalancedMove(debitMoveLine);
+          boolean reconcileByBalancedAccount =
+              reconcileMethodSelect
+                      == AccountingBatchRepository.AUTO_MOVE_LETTERING_RECONCILE_BY_BALANCED_ACCOUNT
+                  && debitTotalRemaining.compareTo(creditTotalRemaining) == 0;
```

Comment:
It seems it doesn't work, I have an example where I should get a single reconcileGroup for all these MoveLines:
![image](https://user-images.githubusercontent.com/10849292/219620811-00bef3b3-9db0-4762-8c05-ab4cfe343116.png)